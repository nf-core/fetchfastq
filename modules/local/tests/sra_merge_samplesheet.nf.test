nextflow_process {

    name "Test Process SRA_MERGE_SAMPLESHEET"
    script "modules/local/sra_merge_samplesheet.nf"
    process "SRA_MERGE_SAMPLESHEET"
    tag "sra_merge_samplesheet"
    tag "modules_local"

    test("Should run without failures") {

        when {
            params {
                outdir   = "$outputDir"
            }
            process {
                """
                input[0] = file("$baseDir/modules/local/tests/input/sra_merge_samplesheet/samplesheets/*", checkIfExists: true)
                input[1] = file("$baseDir/modules/local/tests/input/sra_merge_samplesheet/mappings/*", checkIfExists: true)
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() },
            )
        }

    }

}
