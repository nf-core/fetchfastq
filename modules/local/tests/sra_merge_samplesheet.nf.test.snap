{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    "samplesheet.csv:md5,ef756557b0735becd2574c2a3f5840b2"
                ],
                "1": [
                    "id_mappings.csv:md5,07262807636ce4b50102308eabdcf253"
                ],
                "2": [
                    "versions.yml:md5,410006679d5e496d8c55e58e78ca6b34"
                ],
                "mappings": [
                    "id_mappings.csv:md5,07262807636ce4b50102308eabdcf253"
                ],
                "samplesheet": [
                    "samplesheet.csv:md5,ef756557b0735becd2574c2a3f5840b2"
                ],
                "versions": [
                    "versions.yml:md5,410006679d5e496d8c55e58e78ca6b34"
                ]
            }
        ],
        "timestamp": "2023-07-06T13:16:36+0000"
    }
}